Feature/remove toc not included warnings

sipke opened the pull request:
All git submodules currently have files not included in openamp-docs, which results in warning
WARNING: document isn't included in any toctree [toc.not_included]

Most of the files could be integrated into the documentations, with first example given including the "build" information for the Echo Example of the openamp-system-reference.

There are some files harder to integrate and for those recommendation is to filter them out using the toc :hidden: feature which suppresses the warning.

Note, that some changes will be needed for files in submodules also as some of the links there are relative hyperlinks, rather than sphinx references. You will notice under Echo Test Build Information -> Compilation, the link to README.md does not work currently, as it is a relative hyperlink rather than sphinx reference. Solvable, just not done yet in this proposal.

Commit:
index: add explicit hidden toc for any submodule files not to include

there are submodules which have document files just for the repository,
so provide a toctree of hidden files so the warning about missing file
in toc is suppressed.
WARNING: document isn't included in any toctree [toc.not_included]

Signed-off-by: Sipke Vriend <sipke@direktembedded.com>

Commit:
demos:echo: add links to reference build documentation

for ease of reference to build information in the submodule add a
content table to the echo example documentation in the
openamp-system-reference. This will also remove the warning that the file
is not in a toc.
WARNING: document isn't included in any toctree [toc.not_included]

Signed-off-by: Sipke Vriend <sipke@direktembedded.com>

sipke commented on demos/echo.rst:
```diff
@@ -117,3 +117,17 @@ This Echo Test Sample is demonstrated in the following reference implementations
+   :maxdepth: 2
+   :caption: Echo Test Build Information
+
+   ../openamp-system-reference/examples/legacy_apps/examples/echo/README
```
Yes, adjusting readme files to suit inclusion into openamp docs would make sense.
Yes, as mentioned in the PR summary (second last paragraph) some links are broken as they do not use sphinx references. I did not fix those for this draft pull request as was just showing how the toc not included warnings can be corrected as example before proceeding to address all as there are quite some.

Commit:
demos:reference_boards: add links to reference build information

Add the examples/legacy_apps/readme from the openamp-system-reference
repository to the reference board page as the build information applies
to all applications rather than any one demo from legacy_apps.

Signed-off-by: Sipke Vriend <sipke@direktembedded.com>

arnopo commented:
> Reworked this slightly.
> 
> * Changed echo demo to link only to the legacy_apps/examples/echo/README file, and modified that README to embed the build information. See end of the echo [readme file](https://openamp--82.org.readthedocs.build/en/82/demos/echo.html).This embedding only works in readthedocs (ignored by github), so the original link is left in the repository. Unfortunately I could not figure out how to make a reference work for both readthedocs and github.
> * Added reference to the legacy_apps/README under [reference boards](https://openamp--82.org.readthedocs.build/en/82/demos/reference_boards.html) as it provides build info for multiple applications. May not be exactly the right place for this eventually, but for now it ensures the toc.not_included warning is suppressed.
> 
> @arnopo if this is the right direction to start cross linking build info between repositories and readthedocs let me know and I will repeat for other demos.

Look to me the good direction.
@tnmysh , @edmooring , @wmamills : any option on that ?
You can compare 
https://openamp--82.org.readthedocs.build/en/82/demos/echo.html
with https://openamp.readthedocs.io/en/v2025.10.0/demos/echo.html

Commit:
demos:linux_rpc: add references to build documentation in reference repo

for ease of reference to build information in the submodule add a
content table to the example documentation in the
openamp-system-reference. This will also remove the warning that the file
is not in a toc.
WARNING: document isn't included in any toctree [toc.not_included]

Signed-off-by: Sipke Vriend <sipke@direktembedded.com>

Commit:
demos:matrix_multiply: add reference to build documentation in reference

for ease of reference to build information in the submodule add a
content table to the example documentation in the
openamp-system-reference. This will also remove the warning that the file
is not in a toc.
WARNING: document isn't included in any toctree [toc.not_included]

Signed-off-by: Sipke Vriend <sipke@direktembedded.com>

Commit:
demos:rpmsg_multi_services: add links to reference build documentation

for ease of reference to build information in the submodule add a
content table to the example documentation in the
openamp-system-reference. This will also remove the warning that the file
is not in a toc.
WARNING: document isn't included in any toctree [toc.not_included]

Signed-off-by: Sipke Vriend <sipke@direktembedded.com>

Commit:
demos:linux_rpc: add reference to build of remote processor rpc demo

the readthedocs demo is for linux interprocess example of rpc, but there
is a remote side app as well in reference repository so add reference to
it at the end of the demo for those wishing to try it.

Commit:
demos: add amd specific system reference info to reference boards page

Integrated the openamp-system-reference repository readme pages with
amd specific setup and build information.

Signed-off-by: Sipke Vriend <sipke@direktembedded.com>

Commit:
index: suppress warnings for readme that do not have readthedocs example

These readme pages do not have a description page in readthedocs currently,
so suppress these until decided whether to add somewhere in readthedocs
where they can be integrated.

Signed-off-by: Sipke Vriend <sipke@direktembedded.com>

Commit:
conf: do not include hidden tocs in side bar

by default all toc entries via toctree:: are included into the sidebar,
including hidden ones which spams the menu, so set includehidden to
false in the theme options.

Signed-off-by: Sipke Vriend <sipke@direktembedded.com>

Commit:
demos: integrate readmes from hypervisorless_virtio_zcu102

Integrate the build and reference documentation from the
hypervisorless virtio repository into the demo.

Signed-off-by: Sipke Vriend <sipke@direktembedded.com>

Commit:
index: hide the open-amp and library readme files

the open-amp and libmetal repository doxygen documentation, including
readme files are included via the reference link in the navigation bar
on the left, so list them as hidden in the main index.

Signed-off-by: Sipke Vriend <sipke@direktembedded.com>

Commit:
index: hide the openamp-docs readme file

the readme file for openamp-docs is only for how to generate these docs
so hide it to avoid invalid warning that it is missing from table of
contents.

Signed-off-by: Sipke Vriend <sipke@direktembedded.com>

Commit:
index: hide maintainers file from table of contents

Proposing to include these maintainers files through the reference link
in the navigation bar on the left, whence we can hid them from the
table of contents in the openamp-docs repository.
Two additional pull requests would be made to open-amp and libmetal to
support this.

Signed-off-by: Sipke Vriend <sipke@direktembedded.com>

Commit:
index: hide lopper repository specific documentation files

hide from table of contenst the readme files in the lopper repository
which detail what is in that repository and how to test rather than
are actual documentation. This removes the toc.notincluded warning as
that content isn't inside any readthedocs documentation.

Signed-off-by: Sipke Vriend <sipke@direktembedded.com>

Commit:
tools: rearrange lopper files and intergrate docs from lopper repo

to enable the integration of the documentation in the lopper repository
rearrange the files and tocs to make lopper a top leavel tool and add
the repository documentation files as table of content entries.

Signed-off-by: Sipke Vriend <sipke@direktembedded.com>

Commit:
Remove invalid readme file

remove non existent openamp-docs/README as it is already in there as
README at base directory.

Signed-off-by: Sipke Vriend <sipke@direktembedded.com>

arnopo reviewed (commented):
I haven’t spent much time on the review since it is still in draft, but the rework seems pretty good to me.
Pleas find 2 comments .

Commit:
overview: remove duplicate reference and invalid indent warnings

two references were labelled Virtio so change second to Virtio Driver to
avoid the warning.
WARNING: Duplicate explicit target name: "virtio".
Add new lines after bullet point sections to avoid error.
ERROR: Unexpected indentation. [docutils]

Signed-off-by: Sipke Vriend <sipke@direktembedded.com>

sipke commented on index.rst:
```diff
@@ -34,9 +34,11 @@ Welcome to the OpenAMP Project Documentation
    openamp-docs/README
    open-amp/.github/actions/build_ci/README
    open-amp/README
+   open-amp/MAINTAINERS
```
Interesting. doxygen looks to have generated two files. The one you link is empty, the other has the content.
```
find _build/openamp/doc/html/ | grep -i m_a_i_n_t_a_i_n_e_r_s
_build/openamp/doc/html/md__m_a_i_n_t_a_i_n_e_r_s.html
_build/openamp/doc/html/_m_a_i_n_t_a_i_n_e_r_s_8md.html
```

The one with the expected content can be navigated to via REFERENCE, open-amp API, Maintainers.
The page you linked I found via the search. Is that how you got to it, or is it referenced in a menu item also?
Do not know yet why the two files.

arnopo commented:
Hi @sipke,
Please, could you rebase it  to generate doc including submodule update?   I guess thatw e can now remove the draft, right?

sipke commented:
With submodule PRs in main, rebased to it so this PR can be reviewed.

arnopo reviewed (approved):
LGTM
